### ISSUE TITLE
camera parameters

### ISSUE THREAD

**Issue by li1u** [facebookresearch/mvdust3r#41]
Excuse me, I see that the camera parameters in your code are adjusted according to the size of the camera. I would like to ask if I want to use the generated camera parameters for reconstruction in other algorithms, not just visualization, what should I do?

**Comment by li1u** [comment]
The main problem is that the camera parameters obtained by other methods are completely wrong when reconstructed. My focal is obtained by estimate_focal_knowing_depth, and c2w is obtained by calibrate_camera_pnpransac.

`focals = estimate_focal_knowing_depth(pts3d[0][None].cuda(), valid_first[None].cuda()).cpu().item()

        intrinsics = torch.eye(3,)
        intrinsics[0, 0] = focals
        intrinsics[1, 1] = focals
        intrinsics[0, 2] = w / 2
        intrinsics[1, 2] = h / 2
        intrinsics = intrinsics.cuda()

        focals = torch.Tensor([focals]).reshape(1,).repeat(len(rgbimg))

        y_coords, x_coords = torch.meshgrid(torch.arange(h), torch.arange(w), indexing='ij')
        pixel_coords = torch.stack([x_coords, y_coords], dim=-1).float().cuda()  # [H, W, 2]

        c2ws = []
        for (pr_pt, valid) in zip(pts3d, msk):
            c2ws_i = calibrate_camera_pnpransac(pr_pt.cuda().flatten(0,1)[None], pixel_coords.flatten(0,1)[None], valid.cuda().flatten(0,1)[None], intrinsics[None])
            c2ws.append(c2ws_i[0])

        cams2world = torch.stack(c2ws, dim=0).cpu()  # [N, 4, 4]
        focals = to_numpy(focals)`

![Image](https://github.com/user-attachments/assets/e1ecd43e-ac62-48d3-b198-15e870cd7f07)
As shown in the picture, we actually shot around the object, but the effect shown is that the camera is in the center. Have you used the generated camera parameters?
